components/exercises/chapter-4-oscar/AList/CityList.tsx

```diff
@@ -0,0 +1,138 @@
+        </div>
+      ) : (
+        <details className='mb-2'>
+          <summary className='cursor-pointer rounded-lg bg-yellow-600 p-3 shadow'>
+            <span className='font-semibold'>
+              {item.rest && `${item.rest.value} and the rest of the cities`}
+            </span>
+          </summary>
+          <div className='bg-white p-4'>
+            {item.rest ? <ListItem item={item.rest} /> : null}
+          </div>
+        </details>
+      )}
+    </li>
+  );
+}
+
+export const cityGen = cityGenerator();
+
+/**
+ * It displays a list of cities and their corresponding main food.
+ * It allows adding new cities to the list.
+ */
+export default function CityList() {
+  const [items, setItems] = useState(initialItems);
+
+  /**
+   * It adds a new city.
+   */
+  function handleClick(): void {
+    if (items) {
+      const itemsArray = listToArray(items);
+
+      // Check if the current city is already in the list
+      const currentCity = cityGen(); // Get the next city in the sequence
+      const cityAlreadyExists = itemsArray.includes(currentCity);
+
+      // If the city is not in the list, add it
+      if (!cityAlreadyExists) {
+        setItems((prev) => prepend(currentCity, prev));
+      }
+    }
+  }
+
+  return (
+    <div className='flex'>
+      <div className='max-w-lg flex-[1.8] p-4'>
+        <details className='mb-2'>
+          <summary className='mb-4 cursor-pointer rounded-lg bg-yellow-300 p-4 shadow-md'>
+            <span className='font-semibold'>Cities I would like to visit</span>
+          </summary>
+          <ul className='ml-8 max-h-[366px] space-y-4 overflow-hidden'>
+            {items && <ListItem item={items} />}
+          </ul>
+        </details>
+        <p className=' mb-2 font-semibold'>
+          Countries of the listed cities above
+        </p>
+        <div className='flex gap-2'>
+          <div className='flex h-[200px] flex-1 flex-col'>
+            {listToArray(items).map((item) => {
+              if (item === 'Paris') {
+                return <div key={item.toString()}>France 🇫🇷</div>;
+              } else if (item === 'London') {
+                return <div key={item.toString()}>United Kingdom 🇬🇧</div>;
+              } else if (item === 'Barcelona') {
+                return <div key={item.toString()}>Spain 🇪🇸</div>;
+              } else if (item === 'Porto') {
+                return <div key={item.toString()}>Portugal 🇵🇹</div>;
+              } else if (item === 'Milan') {
+                return <div key={item.toString()}>Italy 🇮🇹</div>;
+              } else if (item === 'Geneva') {
+                return <div key={item.toString()}>Switzerland 🇨🇭</div>;
+              } else if (item === 'Budapest') {
+                return <div key={item.toString()}>Hungary 🇭🇺</div>;
+              } else {
+                return <div key={item?.toString()}>No city decided</div>;
+              }
+            })}
+            {listToArray(items).length === 7 ? (
```

Comment:
If the list of cities changes this conditional will stop working. Better to make this dynamic:
```
{listToArray(items).length === cities.length ? (
```

Reply:
changed, thank you